Dev Services for Kafka combined with continuous testing not working right.

### Describe the bug

I'm trying to combine Dev Services for Kafka with Continuous testing and the experience isn't that great. In all the scenarios I mention below I am creating an application from scratch using the SmallRye Reactive messaging & RESTEasy reactive extensions.

In all scenarios mentioned below the continuous testing hangs and the entire `quarkus:dev` needs to be `CTRL-C`'ed and restarted.

The `PriceResourceTests` class mentioned in all scenarios looks like this:
```java
@QuarkusTest
class PriceResourceTests {
    @TestHTTPEndpoint(PriceResource.class)
    @TestHTTPResource("/stream")
    URI uri;

    @Test
    public void sseStream() {
        Client client = ClientBuilder.newClient();
        WebTarget target = client.target(this.uri);

        List<Double> received = new CopyOnWriteArrayList<>();

        try (SseEventSource source = SseEventSource.target(target).build()) {
            source.register(inboundSseEvent -> received.add(Double.valueOf(inboundSseEvent.readData())));
            source.open();

            await()
                .atMost(Duration.ofSeconds(15))
                .until(() -> received.size() == 2);
        }

        assertThat(received)
            .hasSize(2)
            .allMatch(value -> (value >= 0) && (value < 100));
    }
}
```

When completed, the `PriceGenerator` class looks like this:
```java
@ApplicationScoped
public class PriceGenerator {
    private final Random random = new Random();

    @Outgoing("generated-price")
    public Multi<Integer> generate() {
        return Multi.createFrom().ticks().every(Duration.ofSeconds(5))
                .onOverflow().drop()
                .map(tick -> this.random.nextInt(100));
    }
}
```

When completed, the `PriceConverter` class looks like this:
```java
@ApplicationScoped
public class PriceConverter {
    private static final double CONVERSION_RATE = 0.88;

    @Incoming("prices")
    @Outgoing("processed-prices")
    @Broadcast
    public double process(int priceInUsd) {
        return priceInUsd * CONVERSION_RATE;
    }
}
```

When completed, the `PriceResource` class looks like this:
```java
@Path("/prices")
public class PriceResource {
    private final Publisher<Double> processedPrices;

    public PriceResource(@Channel("processed-prices") Publisher<Double> processedPrices) {
        this.processedPrices = processedPrices;
    }

    @GET
    @Path("/stream")
    @Produces(MediaType.SERVER_SENT_EVENTS)
    public Publisher<Double> ssePrices() {
        return this.processedPrices;
    }
}
```

When completed, the `application.properties` file looks like this:
```properties
# Configure the Kafka sink (we write to it)
mp.messaging.outgoing.generated-price.connector=smallrye-kafka
mp.messaging.outgoing.generated-price.topic=prices
mp.messaging.outgoing.generated-price.value.serializer=org.apache.kafka.common.serialization.IntegerSerializer

# Configure the Kafka source (we read from it)
mp.messaging.incoming.prices.connector=smallrye-kafka
mp.messaging.incoming.prices.health-readiness-enabled=false
mp.messaging.incoming.prices.topic=prices
mp.messaging.incoming.prices.value.deserializer=org.apache.kafka.common.serialization.IntegerDeserializer
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

# Scenario 1
1. Generate project
2. Add these dependencies to `pom.xml`
   ```xml
   <dependency>
      <groupId>org.jboss.resteasy</groupId>
      <artifactId>resteasy-client</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>io.rest-assured</groupId>
      <artifactId>rest-assured</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>org.awaitility</groupId>
      <artifactId>awaitility</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>org.assertj</groupId>
      <artifactId>assertj-core</artifactId>
      <scope>test</scope>
    </dependency>
   ```
3. Add an empty `PriceGenerator`, `PriceConverter`, and `PriceResource` classes in `src/main/java/com/example`
4. Add a completed `PriceResourceTests` class in `src/test/java/com/example`
5. Run `quarkus:dev`
6. Turn on testing
7. Complete `PriceResource` class
8. Complete `PriceGenerator` class
9. Complete `PriceConverter` class
10. Complete `application.properties`

At this step the console output indicates the test is running but it never completes, even though the test itself has a 15 second timeout. Furthermore, it seems the application itself never even restarts itself unless, from another terminal, issue an http request to `/prices/stream`. Still, the continuous testing is hung and I have to CTRL-C.

# Scenario 2
1. Generate project
2. Add these dependencies to `pom.xml`
   ```xml
   <dependency>
      <groupId>org.jboss.resteasy</groupId>
      <artifactId>resteasy-client</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>io.rest-assured</groupId>
      <artifactId>rest-assured</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>org.awaitility</groupId>
      <artifactId>awaitility</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>org.assertj</groupId>
      <artifactId>assertj-core</artifactId>
      <scope>test</scope>
    </dependency>
   ```
3. Add an empty `PriceGenerator`, `PriceConverter`, and `PriceResource` classes in `src/main/java/com/example`
4. Add a completed `PriceResourceTests` class in `src/test/java/com/example`
5. Complete `application.properties`
6. Run `quarkus:dev`
7. Turn on testing
8. Complete `PriceResource` class
9. Complete `PriceGenerator` class
10. Complete `PriceConverter` class

Same result as scenario 1

# Scenario 3
1. Generate project
2. Add these dependencies to `pom.xml`
   ```xml
   <dependency>
      <groupId>org.jboss.resteasy</groupId>
      <artifactId>resteasy-client</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>io.rest-assured</groupId>
      <artifactId>rest-assured</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>org.awaitility</groupId>
      <artifactId>awaitility</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>org.assertj</groupId>
      <artifactId>assertj-core</artifactId>
      <scope>test</scope>
    </dependency>
   ```
3. Add an empty `PriceGenerator`, `PriceConverter`, and `PriceResource` classes in `src/main/java/com/example`
4. Add a completed `PriceResourceTests` class in `src/test/java/com/example`
5. Complete `application.properties`
6. Run `quarkus:dev`
7. Complete `PriceResource` class
8. Complete `PriceGenerator` class
9. Complete `PriceConverter` class
10. Turn on testing

Same result as scenario 1

# Scenario 4
1. Generate project
2. Add these dependencies to `pom.xml`
   ```xml
   <dependency>
      <groupId>org.jboss.resteasy</groupId>
      <artifactId>resteasy-client</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>io.rest-assured</groupId>
      <artifactId>rest-assured</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>org.awaitility</groupId>
      <artifactId>awaitility</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>org.assertj</groupId>
      <artifactId>assertj-core</artifactId>
      <scope>test</scope>
    </dependency>
   ```
3. Add a completed `PriceGenerator` and `PriceResource` classes in `src/main/java/com/example`
4. Add a completed `PriceResourceTests` class in `src/test/java/com/example`
5. Complete `application.properties`
6. Run `quarkus:dev`
7. Turn on testing
10. Complete `PriceConverter` class

Same result as scenario 1

### Output of `uname -a` or `ver`

`Darwin edeandre-mac 20.5.0 Darwin Kernel Version 20.5.0: Sat May  8 05:10:33 PDT 2021; root:xnu-7195.121.3~9/RELEASE_X86_64 x86_64`

### Output of `java -version`
```
openjdk version "11.0.2" 2019-01-15
OpenJDK Runtime Environment 18.9 (build 11.0.2+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)
```

### GraalVM version (if different from Java)
```
openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment GraalVM CE 21.1.0 (build 11.0.11+8-jvmci-21.1-b05)
OpenJDK 64-Bit Server VM GraalVM CE 21.1.0 (build 11.0.11+8-jvmci-21.1-b05, mixed mode, sharing)
```

### Quarkus version or git rev

`2.1.0.Final`

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
```
Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d)
Maven home: /Users/edeandre/.m2/wrapper/dists/apache-maven-3.8.1-bin/2l5mhf2pq2clrde7f7qp1rdt5m/apache-maven-3.8.1
Java version: 11.0.2, vendor: Oracle Corporation, runtime: /Users/edeandre/.sdkman/candidates/java/11.0.2-open
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.16", arch: "x86_64", family: "mac"
```

### Additional information

_No response_

@stuartwdouglas I'm not able to build that PR branch. I've tried twice now and it keeps failing:

```
[INFO] Quarkus - Integration Tests - Redis DevService ..... SUCCESS [  0.940 s]
[INFO] Quarkus - Integration Tests - gRPC - TLS ........... SUCCESS [  4.754 s]
[INFO] Quarkus - Integration Tests - gRPC - Plain Text with gzip compression SUCCESS [  5.384 s]
[INFO] Quarkus - Integration Tests - gRPC - Plain Text with Mutiny SUCCESS [  4.340 s]
[INFO] Quarkus - Integration Tests - gRPC - TLS with mutual auth SUCCESS [  3.549 s]
[INFO] Quarkus - Integration Tests - gRPC - Streaming ..... FAILURE [  2.533 s]
[INFO] Quarkus - Integration Tests - gRPC - Interceptors .. SKIPPED
[INFO] Quarkus - Integration Tests - gRPC - Proto v2 ...... SKIPPED
[INFO] Quarkus - Integration Tests - gRPC - Health ........ SKIPPED
[INFO] Quarkus - Integration Tests - gRPC - Hibernate ..... SKIPPED
[INFO] Quarkus - Integration Tests - gRPC - Hibernate Reactive SKIPPED
[INFO] Quarkus - Integration Tests - Google Cloud Functions HTTP SKIPPED
[INFO] Quarkus - Integration Tests - Google Cloud Function  SKIPPED
[INFO] Quarkus - Documentation ............................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  27:03 min
[INFO] Finished at: 2021-08-11T16:03:22-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:generate-code (default) on project quarkus-integration-test-grpc-streaming: Quarkus code generation phase has failed: InvocationTargetException: Failed to generate Java classes from proto files: [/Users/edeandre/workspaces/quarkus/quarkus/integration-tests/grpc-streaming/src/main/proto/streaming.proto] to /Users/edeandre/workspaces/quarkus/quarkus/integration-tests/grpc-streaming/target/generated-sources/grpc -> [Help 1]
```

--- a/extensions/smallrye-reactive-messaging-kafka/deployment/pom.xml
+++ b/extensions/smallrye-reactive-messaging-kafka/deployment/pom.xml
@@ -60,12 +60,43 @@
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-jsonb-deployment</artifactId>
+            <artifactId>quarkus-resteasy-jsonb-deployment</artifactId>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>io.apicurio</groupId>
             <artifactId>apicurio-registry-serdes-avro-serde</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-junit5</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-rest-client-deployment</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-apicurio-registry-avro-deployment</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.rest-assured</groupId>
+            <artifactId>rest-assured</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.awaitility</groupId>
+            <artifactId>awaitility</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-vertx-http-deployment</artifactId>
+            <type>test-jar</type>
             <scope>test</scope>
         </dependency>
     </dependencies>
@@ -84,7 +115,33 @@
                     </annotationProcessorPaths>
                 </configuration>
             </plugin>
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <skip>true</skip>
+                </configuration>
+            </plugin>
         </plugins>
     </build>
 
+    <profiles>
+        <profile>
+            <id>test-kafka</id>
+            <activation>
+                <property>
+                    <name>start-containers</name>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <skip>false</skip>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
 </project>

--- a/extensions/smallrye-reactive-messaging-kafka/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/testing/KafkaDevServicesContinuousTestingTestCase.java
+++ b/extensions/smallrye-reactive-messaging-kafka/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/testing/KafkaDevServicesContinuousTestingTestCase.java
@@ -1,0 +1,91 @@
+package io.quarkus.smallrye.reactivemessaging.kafka.deployment.testing;
+
+import java.util.function.Supplier;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusDevModeTest;
+import io.quarkus.vertx.http.testrunner.ContinuousTestingTestUtils;
+
+public class KafkaDevServicesContinuousTestingTestCase {
+
+    static final String FINAL_APP_PROPERTIES = ContinuousTestingTestUtils.appProperties(
+            "mp.messaging.outgoing.generated-price.connector=smallrye-kafka",
+            "mp.messaging.outgoing.generated-price.topic=prices",
+            "mp.messaging.outgoing.generated-price.value.serializer=org.apache.kafka.common.serialization.IntegerSerializer",
+            "mp.messaging.incoming.prices.connector=smallrye-kafka",
+            "mp.messaging.incoming.prices.health-readiness-enabled=false",
+            "mp.messaging.incoming.prices.topic=prices",
+            "mp.messaging.incoming.prices.value.deserializer=org.apache.kafka.common.serialization.IntegerDeserializer");
+
+    @RegisterExtension
+    public static QuarkusDevModeTest test = new QuarkusDevModeTest()
+            .setArchiveProducer(new Supplier<JavaArchive>() {
+                @Override
+                public JavaArchive get() {
+                    return ShrinkWrap.create(JavaArchive.class)
+                            .addClasses(PriceConverter.class, PriceResource.class, PriceGenerator.class)
+                            .addAsResource(new StringAsset(ContinuousTestingTestUtils.appProperties("")),
+                                    "application.properties");
+                }
+            }).setTestArchiveProducer(new Supplier<JavaArchive>() {
+                @Override
+                public JavaArchive get() {
+                    return ShrinkWrap.create(JavaArchive.class).addClass(PriceResourceET.class);
+                }
+            });
+
+    //see https://github.com/quarkusio/quarkus/issues/19180
+    @Test
+    public void testContinuousTestingScenario1() {
+        ContinuousTestingTestUtils utils = new ContinuousTestingTestUtils();
+        var result = utils.waitForNextCompletion();
+        Assertions.assertEquals(0, result.getTotalTestsPassed());
+        Assertions.assertEquals(1, result.getTotalTestsFailed());
+        test.modifySourceFile(PriceResource.class, s -> s.replaceAll("//", ""));
+        result = utils.waitForNextCompletion();
+        Assertions.assertEquals(0, result.getTotalTestsPassed());
+        Assertions.assertEquals(1, result.getTotalTestsFailed());
+        test.modifySourceFile(PriceConverter.class, s -> s.replaceAll("//", ""));
+        result = utils.waitForNextCompletion();
+        Assertions.assertEquals(0, result.getTotalTestsPassed());
+        Assertions.assertEquals(1, result.getTotalTestsFailed());
+        test.modifySourceFile(PriceGenerator.class, s -> s.replaceAll("//", ""));
+        result = utils.waitForNextCompletion();
+        Assertions.assertEquals(0, result.getTotalTestsPassed());
+        Assertions.assertEquals(1, result.getTotalTestsFailed());
+        test.modifyResourceFile("application.properties", s -> FINAL_APP_PROPERTIES);
+        result = utils.waitForNextCompletion();
+        Assertions.assertEquals(1, result.getTotalTestsPassed());
+        Assertions.assertEquals(0, result.getTotalTestsFailed());
+    }
+
+    @Test
+    public void testContinuousTestingScenario2() {
+        ContinuousTestingTestUtils utils = new ContinuousTestingTestUtils();
+        var result = utils.waitForNextCompletion();
+        Assertions.assertEquals(0, result.getTotalTestsPassed());
+        Assertions.assertEquals(1, result.getTotalTestsFailed());
+        test.modifyResourceFile("application.properties", s -> FINAL_APP_PROPERTIES);
+        result = utils.waitForNextCompletion();
+        Assertions.assertEquals(0, result.getTotalTestsPassed());
+        Assertions.assertEquals(1, result.getTotalTestsFailed());
+        test.modifySourceFile(PriceConverter.class, s -> s.replaceAll("//", ""));
+        result = utils.waitForNextCompletion();
+        Assertions.assertEquals(0, result.getTotalTestsPassed());
+        Assertions.assertEquals(1, result.getTotalTestsFailed());
+        test.modifySourceFile(PriceGenerator.class, s -> s.replaceAll("//", ""));
+        result = utils.waitForNextCompletion();
+        Assertions.assertEquals(0, result.getTotalTestsPassed());
+        Assertions.assertEquals(1, result.getTotalTestsFailed());
+        test.modifySourceFile(PriceResource.class, s -> s.replaceAll("//", ""));
+        result = utils.waitForNextCompletion();
+        Assertions.assertEquals(1, result.getTotalTestsPassed());
+        Assertions.assertEquals(0, result.getTotalTestsFailed());
+    }
+}

--- a/extensions/smallrye-reactive-messaging-kafka/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/testing/KafkaDevServicesContinuousTestingWorkingAppPropsTestCase.java
+++ b/extensions/smallrye-reactive-messaging-kafka/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/testing/KafkaDevServicesContinuousTestingWorkingAppPropsTestCase.java
@@ -1,0 +1,59 @@
+package io.quarkus.smallrye.reactivemessaging.kafka.deployment.testing;
+
+import java.util.function.Supplier;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusDevModeTest;
+import io.quarkus.vertx.http.testrunner.ContinuousTestingTestUtils;
+
+public class KafkaDevServicesContinuousTestingWorkingAppPropsTestCase {
+
+    @RegisterExtension
+    public static QuarkusDevModeTest test = new QuarkusDevModeTest()
+            .setArchiveProducer(new Supplier<JavaArchive>() {
+                @Override
+                public JavaArchive get() {
+                    return ShrinkWrap.create(JavaArchive.class)
+                            .addClasses(PriceConverter.class, PriceResource.class, PriceGenerator.class)
+                            .addAsResource(new StringAsset(KafkaDevServicesContinuousTestingTestCase.FINAL_APP_PROPERTIES),
+                                    "application.properties");
+                }
+            }).setTestArchiveProducer(new Supplier<JavaArchive>() {
+                @Override
+                public JavaArchive get() {
+                    return ShrinkWrap.create(JavaArchive.class).addClass(PriceResourceET.class);
+                }
+            });
+
+    /**
+     * Similar to {@link KafkaDevServicesContinuousTestingTestCase} however it starts with application.properties configued
+     *
+     * see https://github.com/quarkusio/quarkus/issues/19180
+     */
+    @Test
+    public void testContinuousTestingScenario3() {
+        ContinuousTestingTestUtils utils = new ContinuousTestingTestUtils();
+        var result = utils.waitForNextCompletion();
+        Assertions.assertEquals(0, result.getTotalTestsPassed());
+        Assertions.assertEquals(1, result.getTotalTestsFailed());
+        test.modifySourceFile(PriceResource.class, s -> s.replaceAll("//", ""));
+        result = utils.waitForNextCompletion();
+        Assertions.assertEquals(0, result.getTotalTestsPassed());
+        Assertions.assertEquals(1, result.getTotalTestsFailed());
+        test.modifySourceFile(PriceConverter.class, s -> s.replaceAll("//", ""));
+        result = utils.waitForNextCompletion();
+        Assertions.assertEquals(0, result.getTotalTestsPassed());
+        Assertions.assertEquals(1, result.getTotalTestsFailed());
+        test.modifySourceFile(PriceGenerator.class, s -> s.replaceAll("//", ""));
+        result = utils.waitForNextCompletion();
+        Assertions.assertEquals(1, result.getTotalTestsPassed());
+        Assertions.assertEquals(0, result.getTotalTestsFailed());
+    }
+
+}

--- a/extensions/smallrye-reactive-messaging-kafka/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/testing/PriceConverter.java
+++ b/extensions/smallrye-reactive-messaging-kafka/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/testing/PriceConverter.java
@@ -1,0 +1,20 @@
+package io.quarkus.smallrye.reactivemessaging.kafka.deployment.testing;
+
+//
+//import io.smallrye.reactive.messaging.annotations.Broadcast;
+//import org.eclipse.microprofile.reactive.messaging.Incoming;
+//import org.eclipse.microprofile.reactive.messaging.Outgoing;
+//
+//import javax.enterprise.context.ApplicationScoped;
+//
+//@ApplicationScoped
+public class PriceConverter {
+    //    private static final double CONVERSION_RATE = 0.88;
+    //
+    //    @Incoming("prices")
+    //    @Outgoing("processed-prices")
+    //    @Broadcast
+    //    public double process(int priceInUsd) {
+    //        return priceInUsd * CONVERSION_RATE;
+    //    }
+}

--- a/extensions/smallrye-reactive-messaging-kafka/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/testing/PriceGenerator.java
+++ b/extensions/smallrye-reactive-messaging-kafka/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/testing/PriceGenerator.java
@@ -1,0 +1,20 @@
+package io.quarkus.smallrye.reactivemessaging.kafka.deployment.testing;
+
+//import io.smallrye.mutiny.Multi;
+//import org.eclipse.microprofile.reactive.messaging.Outgoing;
+//
+//import javax.enterprise.context.ApplicationScoped;
+//import java.time.Duration;
+//import java.util.Random;
+//
+//@ApplicationScoped
+public class PriceGenerator {
+    //        private final Random random = new Random();
+    //
+    //        @Outgoing("generated-price")
+    //        public Multi<Integer> generate() {
+    //            return Multi.createFrom().ticks().every(Duration.ofMillis(10))
+    //                    .onOverflow().drop()
+    //                    .map(tick -> this.random.nextInt(100));
+    //        }
+}

--- a/extensions/smallrye-reactive-messaging-kafka/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/testing/PriceResource.java
+++ b/extensions/smallrye-reactive-messaging-kafka/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/testing/PriceResource.java
@@ -1,0 +1,26 @@
+package io.quarkus.smallrye.reactivemessaging.kafka.deployment.testing;
+
+//
+//import org.eclipse.microprofile.reactive.messaging.Channel;
+//import org.reactivestreams.Publisher;
+//
+//import javax.ws.rs.GET;
+//import javax.ws.rs.Path;
+//import javax.ws.rs.Produces;
+//import javax.ws.rs.core.MediaType;
+//
+//@Path("/prices")
+public class PriceResource {
+    //    private final Publisher<Double> processedPrices;
+    //
+    //    public PriceResource(@Channel("processed-prices") Publisher<Double> processedPrices) {
+    //        this.processedPrices = processedPrices;
+    //    }
+    //
+    //    @GET
+    //    @Path("/stream")
+    //    @Produces(MediaType.SERVER_SENT_EVENTS)
+    //    public Publisher<Double> ssePrices() {
+    //        return this.processedPrices;
+    //    }
+}

--- a/extensions/smallrye-reactive-messaging-kafka/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/testing/PriceResourceET.java
+++ b/extensions/smallrye-reactive-messaging-kafka/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/testing/PriceResourceET.java
@@ -1,0 +1,47 @@
+package io.quarkus.smallrye.reactivemessaging.kafka.deployment.testing;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.sse.SseEventSource;
+
+import org.assertj.core.api.Assertions;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.common.http.TestHTTPEndpoint;
+import io.quarkus.test.common.http.TestHTTPResource;
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+class PriceResourceET {
+    @TestHTTPEndpoint(PriceResource.class)
+    @TestHTTPResource("/stream")
+    URI uri;
+
+    @Test
+    public void sseStream() {
+        Client client = ClientBuilder.newClient();
+        WebTarget target = client.target(this.uri);
+
+        List<Double> received = new CopyOnWriteArrayList<>();
+
+        try (SseEventSource source = SseEventSource.target(target).build()) {
+            source.register(inboundSseEvent -> received.add(Double.valueOf(inboundSseEvent.readData())));
+            source.open();
+
+            Awaitility.await()
+                    .atMost(Duration.ofSeconds(1))
+                    .until(() -> received.size() >= 2);
+        }
+
+        Assertions.assertThat(received)
+                .hasSizeGreaterThan(2)
+                .allMatch(value -> (value >= 0) && (value < 100));
+    }
+}

